applications/luci-app-xfrpc/htdocs/luci-static/resources/view/xfrpc.js

```diff
@@ -0,0 +1,162 @@
+		s.anonymous = true;
+		s.render = function (section_id) {
+			L.Poll.add(function () {
+				return L.resolveDefault(getServiceStatus()).then(function(res) {
+					var view = document.getElementById("service_status");
+					view.innerHTML = renderStatus(res);
+				});
+			});
+
+			return E('div', { class: 'cbi-map' },
+				E('fieldset', { class: 'cbi-section'}, [
+					E('p', { id: 'service_status' },
+						_('Collecting data ...'))
+				])
+			);
+		}
+
+		s = m.section(form.NamedSection, 'common', 'xfrpc');
+		s.dynamic = true;
+
+		s.tab('common', _('Common Settings'));
+		s.tab('init', _('Startup Settings'));
+
+		o = s.taboption('common', form.Value, 'server_addr', _('Server address'), 
+			'%s <br /> %s'.format(_('Server address specifies the address of the server to connect to.'), 
+			_('By default, this value is "0.0.0.0".')));
+		o.datatype = 'host';
+
+		o = s.taboption('common', form.Value, 'server_port', _('Server port'), 
+			'%s <br /> %s'.format(_('Server port specifies the port to connect to the server on.'),
+			_('By default, this value is 7000.')));
+		o.datatype = 'port';
+
+		o = s.taboption('common', form.Value, 'token', _('Token'),
+			'%s <br /> %s'.format(_('Token specifies the authorization token used to create keys to be \
+			sent to the server. The server must have a matching token for authorization to succeed.'), 
+			_('By default, this value is "".')));
+
+		o = s.taboption('init', form.SectionValue, 'init', form.TypedSection, 
+			'xfrp', _('Startup Settings'));
+		s = o.subsection;
+		s.anonymous = true;
+		s.dynamic = true;
+
+		o = s.option(form.Flag, 'disabled', _('Disabled xfrpc service'));
+		o.datatype = 'bool';
+		o.optional = true;
+
+		o = s.option(form.ListValue, 'loglevel', _('Log level'), 
+			'%s <br /> %s'.format(_('LogLevel specifies the minimum log level. Valid values are "Debug", "Info", \
+			"Notice", "Warning", "Error", "Critical", "Alert" and "Emergency".'),
+			_('By default, this value is "Info".')));
+		o.value(8, _('Debug'))
+		o.value(7, _('Info'))
+		o.value(6, _('Notice'))
+		o.value(5, _('Warning'))
+		o.value(4, _('Error'))
+		o.value(3, _('Critical'))
+		o.value(2, _('Alert'))
+		o.value(1, _('Emergency'))
+
+		s = m.section(form.GridSection, 'xfrpc', _('Proxy Settings'));
+		s.addremove = true;
+		s.filter = function(s) { return s !== 'common'; };
+		s.renderSectionAdd = function(extra_class) {
+			var el = form.GridSection.prototype.renderSectionAdd.apply(this, arguments),
+				nameEl = el.querySelector('.cbi-section-create-name');
+			ui.addValidator(nameEl, 'uciname', true, function(v) {
+				if (v === 'common') return _('Name can not be "common"');
+				return true;
+			}, 'blur', 'keyup');
+			return el;
+		}
+
+		s.tab('general', _('General Settings'));
+		s.tab('http', _('HTTP Settings'));
+
+		s.option(form.Value, 'type', _('Proxy type'));
+		s.option(form.Value, 'local_ip', _('Local IP'));
+		s.option(form.Value, 'local_port', _('Local port'));
```

Comment:
What about this? I think datatypes also make sense here

Reply:
@feckert  when I comment that 3 lines, which will bring the effect as the pic show:
![image](https://user-images.githubusercontent.com/1182593/176337634-0ac30c6e-97f3-4ee8-8ac4-5d064bb73355.png)

when I add new proxy setting, the restrict of type ipaddr and port has effect. therefore I think no need to add datatype there.

thanks for your patient reveiw.

Reply:
Sorry my mistake I read it wrong in a hurry.